### DISCUSSION TITLE
Journal of Open Source Software

### DISCUSSION THREAD

**Discussion by Dennis1989** [online-ml/river#1103] [General]
Hey there, 

I would like to raise your attention to JOSS. 

https://joss.theoj.org

Did you consider publishing river in such an open-source software journal? This could boost the prominence of the algorithm.

Best,

Dennis

**Comment by MaxHalford** [comment]
Hey Dennis :)

We've published to JMLR's software track, but not to JOSS. Do you think publishing to JOSS would be beneficial? The upside I see is that we update the list of contributors on the package. The downside is that there would be two papers people could cite.

**Reply by Dennis1989** [reply]
Hey Max, 

Yes, I was aware of the JMLR publication. I was considering JOSS to maybe reach a wider audience (not only ml scholars). I think one advantage of the river framework is that the stream algorithms can actually be used not only to create benchmarking studies but also to use them in a more applied way for various research endeavors (e.g., computational social scientists who want to monitor conversation streams over time).

But yeah, you are right. The downside is that there are potentially two papers to cite then.

**Reply by MaxHalford** [reply]
I think there is value in highlighting parts of River. There's a lot of stuff of paper, of a generalist paper like the JMLR one hide a lot of the detail. I think it could be worthwhile to publish a separate paper highlighting the clustering module. As you say, we could emphasize practical aspects, such as built-in tools for monitoring streams (Twitch, Twitter). Also, I think @hoanganhngo610 might want to publish such a paper too. I think it would be nice to have all our names on a paper :)

That being said, I do think the cluster module needs more work done to it before we can be comfortable publishing a paper about it. For instance, I still haven't put any time into refactoring textClust. Also, I think we're lacking a bit of user guide documentation. However, this paper might be the extra incentive we need :)
